src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -156,6 +156,56 @@ RankedTensorType getReductionOutputType(RankedTensorType operandTy,
   return RankedTensorType::get(dims, operandTy.getElementType());
 }
 
+// Reduction with axes is from ConstantOp.
+// Only ReduceSum call this function now.
+RankedTensorType getReductionOutputType(RankedTensorType operandTy,
+    DenseElementsAttr axesAttrs, uint64_t keepdims,
+    uint64_t noop_with_empty_axes) {
+  int64_t rank = operandTy.getRank();
+
+  SmallVector<int64_t, 4> axes;
+  if (axesAttrs) {
+    for (auto element : axesAttrs.getValues<IntegerAttr>()) {
+      int64_t axis = element.getInt();
+      axis = axis >= 0 ? axis : (rank + axis);
+      assert(axis >= -rank && axis <= rank - 1);
```

Comment:
If this come from the user, I would use an op error, for sure a message.

Reply:
@chentong319 same issue here as above. after correction, axis >=0 && <= rank -1

Reply:
You are right. Fixed. 
Since this code is in a utility function, some extra work is needed. I tested the code with some examples.